src/engine/Match_search_mode.ml

```diff
@@ -554,147 +554,209 @@ let children_explanations_of_xpat (env : env) (xpat : Xpattern.t) : ME.t list =
 
```

Comment:
No changelog!

Reply:
Re to the test that changed, this is indeed a breaking change. I'd hope it doesn't break many rules, but if someone has a couple of `metavariable-pattern`s where it reused the same metavariable, it may stop working. Need to make that clear in the changelog so folks can easily spot what may be causing their rules to break. We could also provide a rule option to disable this behavior, but I'd personally be fine not adding it for now and wait until we get any complain.